### ISSUE TITLE
ProxyCallReplacer is replacing non-proxy-calls with arbitrary calls

### ISSUE THREAD

**Issue by CreateAndInject** [icsharpcode/ILSpy#1443]
[1.zip](https://github.com/icsharpcode/ILSpy/files/2935095/1.zip)

Click RebelBetting.Client.Properties.Settings : 
![image](https://user-images.githubusercontent.com/2529999/53865503-4ca1ff80-402a-11e9-87a7-ddbc140a2096.png)
![image](https://user-images.githubusercontent.com/2529999/53865519-57f52b00-402a-11e9-8e2e-edba0088d8bd.png)
```
Error decompiling RebelBetting.Client.Properties.Settings.Init
 ---> System.ArgumentOutOfRangeException: 
 index
   at System.ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument argument, ExceptionResource resource) in offset 46
   at System.SZArrayHelper.get_Item[T](Int32 index) in offset 13
   at ICSharpCode.Decompiler.CSharp.CallBuilder.BuildArgumentList(ExpectedTargetDetails expectedTargetDetails, ResolveResult target, IMethod method, Int32 firstParamIndex, IReadOnlyList`1 callArguments, IReadOnlyList`1 argumentToParameterMap) in offset 251
   at ICSharpCode.Decompiler.CSharp.CallBuilder.Build(OpCode callOpCode, IMethod method, IReadOnlyList`1 callArguments, IReadOnlyList`1 argumentToParameterMap, IType constrainedTo) in offset 256
   at ICSharpCode.Decompiler.CSharp.CallBuilder.Build(CallInstruction inst) in offset 382
   at ICSharpCode.Decompiler.CSharp.ExpressionBuilder.VisitCall(Call inst, TranslationContext context) in offset 29
   at ICSharpCode.Decompiler.IL.Call.AcceptVisitor[C,T](ILVisitor`2 visitor, C context) in offset 9
   at ICSharpCode.Decompiler.CSharp.ExpressionBuilder.Translate(ILInstruction inst, IType typeHint) in offset 61
   at ICSharpCode.Decompiler.CSharp.StatementBuilder.Default(ILInstruction inst) in offset 14
   at ICSharpCode.Decompiler.IL.ILVisitor`1.VisitCall(Call inst) in offset 1
   at ICSharpCode.Decompiler.IL.Call.AcceptVisitor[T](ILVisitor`1 visitor) in offset 1
   at ICSharpCode.Decompiler.CSharp.StatementBuilder.Convert(ILInstruction inst) in offset 16
   at ICSharpCode.Decompiler.CSharp.StatementBuilder.VisitBlock(Block block) in offset 55
   at ICSharpCode.Decompiler.IL.Block.AcceptVisitor[T](ILVisitor`1 visitor) in offset 1
   at ICSharpCode.Decompiler.CSharp.StatementBuilder.Convert(ILInstruction inst) in offset 16
   at ICSharpCode.Decompiler.CSharp.StatementBuilder.VisitIfInstruction(IfInstruction inst) in offset 27
   at ICSharpCode.Decompiler.IL.IfInstruction.AcceptVisitor[T](ILVisitor`1 visitor) in offset 1
   at ICSharpCode.Decompiler.CSharp.StatementBuilder.Convert(ILInstruction inst) in offset 16
   at ICSharpCode.Decompiler.CSharp.StatementBuilder.ConvertBlockContainer(BlockStatement blockStatement, BlockContainer container, IEnumerable`1 blocks, Boolean isLoop) in offset 155
   at ICSharpCode.Decompiler.CSharp.StatementBuilder.ConvertBlockContainer(BlockContainer container, Boolean isLoop) in offset 7
   at ICSharpCode.Decompiler.CSharp.StatementBuilder.VisitBlockContainer(BlockContainer container) in offset 168
   at ICSharpCode.Decompiler.IL.BlockContainer.AcceptVisitor[T](ILVisitor`1 visitor) in offset 1
   at ICSharpCode.Decompiler.CSharp.StatementBuilder.Convert(ILInstruction inst) in offset 16
   at ICSharpCode.Decompiler.CSharp.StatementBuilder.ConvertAsBlock(ILInstruction inst) in offset 1
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.DecompileBody(IMethod method, EntityDeclaration entityDecl, DecompileRun decompileRun, ITypeResolveContext decompilationContext) in offset 714
-- continuing with outer exception (ICSharpCode.Decompiler.DecompilerException) --
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.DecompileBody(IMethod method, EntityDeclaration entityDecl, DecompileRun decompileRun, ITypeResolveContext decompilationContext) in offset 1073
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.DoDecompile(IMethod method, DecompileRun decompileRun, ITypeResolveContext decompilationContext) in offset 249
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.DoDecompile(ITypeDefinition typeDef, DecompileRun decompileRun, ITypeResolveContext decompilationContext) in offset 783
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.Decompile(IEnumerable`1 definitions) in offset 304
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.Decompile(EntityHandle[] definitions) in offset 1
   at ICSharpCode.ILSpy.CSharpLanguage.DecompileType(ITypeDefinition type, ITextOutput output, DecompilationOptions options) in CSharpLanguage.cs:line 286
   at ICSharpCode.ILSpy.TreeNodes.TypeTreeNode.Decompile(Language language, ITextOutput output, DecompilationOptions options) in TypeTreeNode.cs:line 108
   at ICSharpCode.ILSpy.TextView.DecompilerTextView.DecompileNodes(DecompilationContext context, ITextOutput textOutput) in DecompilerTextView.cs:line 542
   at ICSharpCode.ILSpy.TextView.DecompilerTextView.<>c__DisplayClass34_0.<DecompileAsync>b__0() in DecompilerTextView.cs:line 521
```

There's no problem in ILSpy3.2

**Comment by CreateAndInject** [comment]
There're asserts also in  RebelBetting.Client.Forms.MainForm.

**Comment by siegfriedpammer** [comment]
> There're asserts also in RebelBetting.Client.Forms.MainForm.

Fixed in 40b6f0c7bf7a776538efab540b82d4ec7f7fdf65.
